app/assets/src/components/views/SampleView/SampleView.jsx

```diff
@@ -408,28 +419,62 @@ class SampleView extends React.Component {
                 options={Object.keys(sampleIdToNameMap).map(sampleId => ({
                   label: sampleIdToNameMap[sampleId],
                   id: sampleId,
-                  onClick: () => window.open(`/samples/${sampleId}`, "_self")
+                  onClick: () => {
+                    window.open(`/samples/${sampleId}`, "_self");
+                    logAnalyticsEvent("SampleView_header-title_clicked", {
+                      sampleId
+                    });
+                  }
                 }))}
               />
               <div className={cs.sampleDetailsLinkContainer}>
                 <span
                   className={cs.sampleDetailsLink}
-                  onClick={this.toggleSampleDetailsSidebar}
+                  onClick={withAnalytics(
+                    this.toggleSampleDetailsSidebar,
+                    "SampleView_sample-details-link_clicked",
+                    {
+                      sampleId: sample.id,
+                      sampleName: sample.name
```

Comment:
out of curiosity, do you always want to log the sample name with the sample ID?

Reply:
In general, I advocate logging the name of an object with the ID as a convenience for analysis downstream. It avoids a JOIN with little cost.